app/code/Magento/Ui/view/base/web/js/grid/masonry.js

```diff
@@ -0,0 +1,261 @@
+                rowHeight = calcHeight < this.maxImageHeight ? calcHeight : this.maxImageHeight;
+                isLastRow = index + 1 === this.rows().length;
+
+                this.assignImagesToRow(rowImages, rowNumber, rowHeight, isLastRow);
+
+                rowImages = [];
+                ratio = 0;
+                rowNumber++;
+
+            }.bind(this));
+        },
+
+        /**
+         * Apply styles, css classes and add properties for images in the row
+         *
+         * @param {Object[]} images
+         * @param {Number} rowNumber
+         * @param {Number} rowHeight
+         * @param {Boolean} isLastRow
+         */
+        assignImagesToRow: function (images, rowNumber, rowHeight, isLastRow) {
+            var imageWidth;
+
+            images.forEach(function (img) {
+                imageWidth = rowHeight * (img.width / img.height).toFixed(2);
+                this.setImageStyles(img, imageWidth, rowHeight);
+                this.setImageClass(img, {
+                    bottom: isLastRow
+                });
+                img.rowNumber = rowNumber;
+            }.bind(this));
+
+            images[0].firstInRow = true;
+            images[images.length - 1].lastInRow = true;
+        },
+
+        /**
+         * Wait for container to initialize
+         */
+        waitForContainer: function (callback) {
+            if (typeof this.container === 'undefined') {
+                setTimeout(function () {
+                    this.waitForContainer(callback);
+                }.bind(this), 500);
+            } else {
+                callback();
+            }
+        },
+
+        /**
+         * Set layout styles when container element is loaded.
+         */
+        setLayoutStylesWhenLoaded: function () {
+            this.waitForContainer(function () {
+                this.setLayoutStyles();
+            }.bind(this));
+        },
+
+        /**
+         * Set styles for every image in layout
+         *
+         * @param {Object} img
+         * @param {Number} width
+         * @param {Number} height
+         */
+        setImageStyles: function (img, width, height) {
+            if (!img.styles) {
+                img.styles = ko.observable();
+            }
+            img.styles({
+                width: parseInt(width, 10) + 'px',
+                height: parseInt(height, 10) + 'px'
+            });
+        },
+
+        /**
+         * Set css classes to and an image
+         *
+         * @param {Object} image
+         * @param {Object} classes
+         */
+        setImageClass: function (image, classes) {
+            if (!image.css) {
+                image.css = ko.observable(classes);
+            }
+            image.css(classes);
+        },
+
+        /**
+         * Set min ratio for images in layout
+         */
+        setMinRatio: function () {
+            if (this.containerWidth <= 640) {
```

Comment:
Better to move these values as a part of the component configuration in defaults?

Reply:
Moved